mu/app.py

```diff
@@ -104,6 +176,34 @@ def excepthook(*exc_args):
     Log exception and exit cleanly.
     """
     logging.error("Unrecoverable error", exc_info=(exc_args))
+    try:
+        root_dir = pathlib.Path(__file__).parent.parent.absolute()
+        icon = str(root_dir / "docs" / "icon_small.png")
+        error = "".join(traceback.format_exception(*exc_args))
+        url = (
+            "https://github.com/mu-editor/mu/issues/new?"
+            "title=Crash%20Report&"
+            "body=Mu%20crashed...%0A%0A```%0ACopy%20error%20here...%0A```"
+        )
+        content = CRASH_TEMPLATE.format(
+            url=url,
+            icon=icon,
+            title=CRASH_TITLE,
+            details=CRASH_DETAILS,
+            copy=CRASH_COPY,
+            log=CRASH_LOG,
+            logfile="file://" + LOG_FILE,
+            error=error,
+            report=CRASH_REPORT,
+            thanks=CRASH_THANKS,
+        )
+        with tempfile.NamedTemporaryFile(
+            suffix=".html", delete=False
+        ) as crash_report:
+            crash_report.write(content.encode("utf-8"))
+        webbrowser.open(crash_report.name)
```

Comment:
I haven't had a chance to look properly to the rest, but as a quick note, in the past I've had issues opening temp files with the browser where in some cases the tmp file was erased soon after handler was closed, so by the time the browser was opened (which can be a really long time forr some users) it couldn't find it any more.

I can't remember exactly which constructor/factory/function from tempfile I was using at the time, I think the (ugly) workaround I implemented was to manually manage the file deletion on a long timer.
Perhaps not closing the handler until the browser is open might be good enough, but we have no idea how long that might be.

Reply:
@carlosperate I've thought about this already.... note the `delete=False` flag when `NamedTemporaryFile` is instantiated... that ensures the file isn't deleted (until, I believe the OS clears out temp files when the user shuts down).

Reply:
Oh yeah, I think I remember looking into that as well, and since the deletion is left to the OS, the files ended up hanging around in my dev machine forever (doing a search found tmp files created by my programme months and months in the past).
I guess we could consider that an OS issue and something outside Mu's scope, but that's the reason I ended up manually deleting the files in my example.